crates/api/src/lib.rs

```
@@ -174,74 +174,77 @@ pub(crate) async fn ban_nonlocal_user_from_local_communities(
   expires: &Option<i64>,
   context: &Data<LemmyContext>,
 ) -> LemmyResult<()> {
-  let ids = Person::list_local_community_ids(&mut context.pool(), target.id).await?;
+  // Only run this code for federated users
+  if !target.local {
+    let ids = Person::list_local_community_ids(&mut context.pool(), target.id).await?;
 
-  for community_id in ids {
-    let expires_dt = check_expire_time(*expires)?;
+    for community_id in ids {
+      let expires_dt = check_expire_time(*expires)?;
 
-    // Ban / unban them from our local communities
-    let community_user_ban_form = CommunityPersonBanForm {
-      community_id,
-      person_id: target.id,
-      expires: Some(expires_dt),
-    };
-
-    if ban {
-      // Ignore all errors for these
-      CommunityPersonBan::ban(&mut context.pool(), &community_user_ban_form)
-        .await
-        .with_lemmy_type(LemmyErrorType::CommunityUserAlreadyBanned)
-        .ok();
-
-      // Also unsubscribe them from the community, if they are subscribed
-      let community_follower_form = CommunityFollowerForm {
+      // Ban / unban them from our local communities
+      let community_user_ban_form = CommunityPersonBanForm {
         community_id,
         person_id: target.id,
-        pending: false,
+        expires: Some(expires_dt),
       };
 
-      CommunityFollower::unfollow(&mut context.pool(), &community_follower_form)
-        .await
-        .ok();
-    } else {
-      CommunityPersonBan::unban(&mut context.pool(), &community_user_ban_form)
-        .await
-        .with_lemmy_type(LemmyErrorType::CommunityUserAlreadyBanned)
-        .ok();
-    }
+      if ban {
+        // Ignore all errors for these
+        CommunityPersonBan::ban(&mut context.pool(), &community_user_ban_form)
+          .await
+          .with_lemmy_type(LemmyErrorType::CommunityUserAlreadyBanned)
```

No reason to add an error message if errors are ignored.